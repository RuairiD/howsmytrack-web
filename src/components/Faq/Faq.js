import React from 'react';

import { List, Typography } from 'antd';

type Props = {};

class Faq extends React.Component<Props> {
    /*
     * Component for displaying FAQ page explaining what FeedbackGroups is etc.
     */

    faqs = [
        {
            question: "What is FeedbackGroups?",
            answer: "FeedbackGroups is a website for musicians to share their work and solicit feedback from eachother. Users may submit one track for feedback per day. Users are then put into groups of 4. Each user can then listen to each other's submitted track and leave feedback. Once a user has left feedback for everyone in their group, the feedback left for their own track will be visible to them.",
        },
        {
            question: "Why does FeedbackGroups exist?",
            answer: "I noticed there were two primary ways musicians would solicit feedback in Discord channels. The first is through some sort of #feedback channel where users can post a track for feedback, but only after providing feedback for the last submitter. I often found this tit-for-tat system a bit crap; there was very little enforcing constructive feedback and often users would leave pretty useless feedback just so they could post their own track. The cycle would continue and nobody would really get any good feedback. The second method was via feedback streams (co-ordinated conference calls where users would play their tracks one-by-one, with feedback to follow). While more effective than tit-for-tat channels, feedback streams are logistically impractical and require a lot of organisation. In addition, there was often an asymmetrical element that made things a bit weird e.g. some 'senior' users having microphone privileges, but not others. FeedbackGroups was designed to take the group mentality of a feedback stream but make it decentralised, avoiding the logistical nonsense, and enforce feedback standards via a rating system.",
        },
        {
            question: "How is my rating calculated?",
            answer: "Users can rate your feedback after reading it. The rating you see when logged in is the mean of all your ratings.",
        },
        {
            question: "Who can see my rating?",
            answer: "Only you. Nobody else can see your rating, and you can't see anybody else's.",
        },
        {
            question: "If nobody can see my rating, what is it used for?",
            answer: "Users with better ratings are grouped together. This (hopefully) incentivises users to write quality feedback to increase the chances of being grouped with other users who write quality feedback in future.",
        },
        {
            question: "Someone gave me feedback I don't agree with. How can I see who wrote it?",
            answer: "You can't, feedback is anonymous. I believe that people leave more honest and more constructive feedback if they are guaranteed anonymity. This may mean you get some feedback you don't like or agree with, but ultimately isn't that the point?",
        },
        {
            question: "Can I add links to my Soundcloud/Spotify/Bandcamp etc. in case people like my stuff?",
            answer: "No. FeedbackGroups is about one thing: feedback. I don't want it being used for anything resembling promotion.",
        },
        {
            question: "How can my feedback be anonymous if my track is linked from my Soundcloud account?",
            answer: "Unfortunately, anonymity cannot be 100% guaranteed if you use tracks from your main Soundcloud account. Feedback is still anonymised within the group though i.e. users can't tell which member of their group left the feedback. You're also welcome to use a burner Soundcloud account if true anonymity is important to you. Furthermore, I hope to add other streaming options in future which would allow true anonymity.",
        }
    ]

    render() {
        return (
            <div>
                <Typography.Title level={2}>FAQ/About FeedbackGroups</Typography.Title>
                <List
                    itemLayout="horizontal"
                    dataSource={this.faqs}
                    renderItem={faq => (
                        <List.Item>
                            <List.Item.Meta
                                title={faq.question}
                                description={faq.answer}
                            />
                        </List.Item>
                    )}
                />
            </div>
        );
    }
}

export default Faq;