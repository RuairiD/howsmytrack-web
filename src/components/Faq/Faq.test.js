import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Faq from "./Faq";

describe("Faq", () => {
    it("displays a list of FAQs", () => {
        const wrapper = shallow(
            <Faq />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
