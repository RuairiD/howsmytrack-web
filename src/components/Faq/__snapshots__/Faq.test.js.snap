// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Faq displays a list of FAQs 1`] = `
<List
  bordered={false}
  dataSource={
    Array [
      Object {
        "answer": <Paragraph>
          <Em
            className=""
          >
            how's my track?
          </Em>
           is a website for musicians to share their work and solicit feedback from each other. Users may submit one track for feedback per day. Users are then put into groups of around four. Each user can then listen to each other's submitted track and leave feedback. Once a user has left feedback for everyone in their group, the feedback left for their own track will be visible to them. 
        </Paragraph>,
        "id": "what-is-hows-my-track",
        "question": "But actually, what is How's My Track?",
      },
      Object {
        "answer": <React.Fragment>
          <Paragraph>
            I noticed there were two primary ways musicians would solicit feedback in Discord channels.
          </Paragraph>
          <Paragraph>
            The first is through some sort of #feedback channel where users can post a track for feedback, but only after providing feedback for the last submitter. I often found this tit-for-tat system a bit crap; there was very little enforcing constructive feedback and often users would leave pretty useless feedback just so they could post their own track. The cycle would continue and nobody would really get any good feedback.
          </Paragraph>
          <Paragraph>
            The second method was via feedback streams (co-ordinated conference calls where users would play their tracks one-by-one, with feedback to follow). While more effective than tit-for-tat channels, feedback streams are logistically impractical and require a lot of organisation. In addition, there was often an asymmetrical element that made things a bit weird e.g. some 'senior' users having microphone privileges, but not others.
          </Paragraph>
          <Paragraph>
            How's My Track? was designed to take the group mentality of a feedback stream but make it decentralised, avoiding the logistical nonsense, and enforce feedback standards via a rating system.
          </Paragraph>
        </React.Fragment>,
        "id": "why-does-hows-my-track-exist",
        "question": "Why does How's My Track? exist?",
      },
      Object {
        "answer": <Paragraph>
          <A
            className=""
            href="/trackurlhelp"
          >
            Check out our guide.
          </A>
        </Paragraph>,
        "id": "how-can-i-get-the-correct-track-url",
        "question": "How can I get the correct track URL for my platform of choice?",
      },
      Object {
        "answer": <React.Fragment>
          <Paragraph>
            Firstly, 
            <Strong
              className=""
            >
              all
            </Strong>
             track submissions are welcome on 
            <Em
              className=""
            >
              how's my track?
            </Em>
            ;if the genre you think best fits your track isn't listed, just pick 
            <Strong
              className=""
            >
              No Genre
            </Strong>
             and your submission will still be accepted and grouped.
          </Paragraph>
          <Paragraph>
            To answer the question, groups can only be formed if 3 or more feedback submissions are received, we restrict genres that don't have enough daily submissions to form at least one group. In the event that genres only receive one submission, a multi-genre group will be formed to ensure that as many people as possible get a group within 24 hours. We also regularly audit submissions to see if it's possible to add more genres, and we're always open to suggestions as well.
          </Paragraph>
        </React.Fragment>,
        "id": "why-is-my-tracks-genre-not-listed",
        "question": "Why is my track's genre not listed on the submission form?",
      },
      Object {
        "answer": <Paragraph>
          Users can rate your feedback after reading it. The rating you see when logged in is the mean of your last 15 ratings, or all of your ratings if you have fewer than than 15 (bear in mind that 15 ratings will be all your ratings from your past 5 groups). User ratings are only re-calculated once per day to obfuscate the source of ratings. 
        </Paragraph>,
        "id": "how-is-my-rating-calculated",
        "question": "How is my rating calculated?",
      },
      Object {
        "answer": <Paragraph>
          Only you. Nobody else can see your rating, and you can't see anybody else's. 
        </Paragraph>,
        "question": "Who can see my rating?",
      },
      Object {
        "answer": <Paragraph>
          Users with better ratings are grouped together. This (hopefully) incentivises users to write quality feedback to increase the chances of being grouped with other users who write quality feedback in future. 
        </Paragraph>,
        "question": "If nobody can see my rating, what is it used for?",
      },
      Object {
        "answer": <Paragraph>
          You can't, feedback is anonymous. I believe that people leave more honest and more constructive feedback if they are guaranteed anonymity. This may mean you get some feedback you don't like or agree with, but ultimately isn't that the point?
        </Paragraph>,
        "question": "Someone gave me feedback I don't agree with. How can I see who wrote it?",
      },
      Object {
        "answer": <Paragraph>
          No. How's My Track? is about one thing: feedback. I don't want it being used for anything resembling promotion. 
        </Paragraph>,
        "question": "Can I add links to my Soundcloud/Spotify/Bandcamp etc. in case people like my stuff?",
      },
      Object {
        "answer": <Paragraph>
          Unfortunately, anonymity cannot be 100% guaranteed if you use tracks from your main Soundcloud account. Feedback is still anonymised within the group though i.e. users can't tell which member of their group left the feedback. You're also welcome to use a burner Soundcloud account if true anonymity is important to you, or use a Google Drive or Dropbox link. 
        </Paragraph>,
        "question": "How can my feedback be anonymous if my track is linked from my Soundcloud account?",
      },
      Object {
        "answer": <Paragraph>
          You're welcome to suggest alternative platforms, but I'll only support platforms that allow embedding of some form.
        </Paragraph>,
        "question": "Why isn't my favourite streaming platform supported?",
      },
    ]
  }
  itemLayout="vertical"
  loading={false}
  pagination={false}
  renderItem={[Function]}
  split={true}
/>
`;
